bitcoin/src/blockdata/witness.rs

```diff
@@ -151,29 +151,30 @@ internal_macros::define_extension_trait! {
             self.push(signature.serialize())
         }
 
-        /// Get Tapscript following BIP341 rules regarding accounting for an annex.
+        /// If this witness is a Taproot script spend, return the leaf script.
         ///
-        /// This does not guarantee that this represents a P2TR [`Witness`]. It
-        /// merely gets the second to last or third to last element depending on
-        /// the first byte of the last element being equal to 0x50.
-        ///
-        /// See [`Script::is_p2tr`] to check whether this is actually a Taproot witness.
+        /// **Deprecated** since this does not return the leaf version. Code that
+        /// assumes a leaf version may be subtly broken once a new Tapscript version
+        /// is deployed.
+        #[deprecated(since = "TBD", note = "use taproot_leaf_script instead")]
```

Comment:
I did read the commit log and PR description but was surprised to see this, I understood the patch was a docs only patch. Are we likely to ever need to point to this change? If not maybe just in the title 'update docs and deprecate' or something